src/captured-globals.js

```diff
@@ -1,16 +1,23 @@
-// Capture prototype to prevent overloading
-function captureGlobals (object) {
-    const newGlobal = object
-    for (const key in object) {
-        const value = object[key]
-        newGlobal[key] = value
-    }
-    for (const key in object.prototype) {
-        const value = object.prototype[key]
-        newGlobal.prototype[key] = value
+import { getInjectionElement } from './utils.js'
+
+let dummyWindow
+if ('document' in globalThis) {
+    const injectionElement = getInjectionElement()
+    // injectionElement is null in some playwright context tests
+    if (injectionElement) {
+        dummyWindow = document.createElement('iframe')
+        injectionElement.appendChild(dummyWindow)
     }
-    return newGlobal
 }
+// Capture prototype to prevent overloading
+function captureGlobal (globalName) {
+    // if we were unable to create a dummy window, return the global
+    // this still has the advantage of preventing aliasing of the global through shawdowing
+    return dummyWindow?.contentWindow?.[globalName] || globalThis[globalName]
```

Comment:
If it really works, it's really neat!
- will the returned values be unaffected by prototype modifications in the current context?
- are there any globals that cannot be copied this way? (i'm thinking `Document`, `Window`, etc)
- can CSP be a problem?
- will removing of the iframe not invalidate the returned objects?

Reply:
1. Yeah it's completely isolated
2. Possibly, I've not tried them all. The globals won't match if that's your concern.
3. Shouldn't be, the URL is about:blank and will be required for basic iframe use. Worst case scenario it's prevented and we fall back to the globalThis.
4. Didn't in my testing, the reference keeps it from being gc'd even once the document goes.